engine.py

```diff
@@ -35,27 +35,59 @@ def __init__(self, *args, **kwargs):
 
         self._ui_created = False
 
-        # set up a very basic logger, assuming it will be overridden
-        self._log = logging.getLogger("tank.tk-shell")
-        self._log.setLevel(logging.INFO)
-        ch = logging.StreamHandler()
-        formatter = logging.Formatter()
-        ch.setFormatter(formatter)
-        self._log.addHandler(ch)
+        self._log = None
```

Comment:
This logic was backwards. We always created a logger and then replaced it if there was one set on the `tk` instance. I've flipped the logic so the logger and `StreamHandler` are only created if necessary.

This means that in the `destroy_engine` and `__del__` methods, if `self._stream_handler` is set, we know that we need to remove it from the logger because it wouldn't have been created in the first place if `tk.log` had been set.